packages/composites/proof-of-reserves/src/utils/protocol.ts

```diff
@@ -23,15 +23,23 @@ export const adapters: AdapterImplementation[] = [
 
 export type Protocol = typeof adapters[number]['NAME']
 
+type AddressData = { token: string; chainId: string; network: string } | AddressList
+
+type AddressList =
+  | { addresses: string[]; chainId: string; network: string }
+  | { addresses: AddressObject[] }
+
+type AddressObject = { address: string; network: string; chainId: string }
+
 // Get address set for protocol
 export const runProtocolAdapter = async (
   jobRunID: string,
   context: AdapterContext,
   protocol: Protocol,
-  data: { token: string; chainId: string; network: string } | { addresses: string[] },
+  data: AddressData,
   config: Config,
 ): Promise<AdapterResponse> => {
-  if (protocol === LIST_ADAPTER) return listAdapter(jobRunID, data as { addresses: string[] })
+  if (protocol === LIST_ADAPTER) return listAdapter(jobRunID, data as AddressList)
```

Comment:
`data` is being coerced to `AddressList` here, but we don't yet know that it is actually that type. That validation happens within the function. We should leave it as so and allow the narrowing within the function to get us to the correct type.